app/views/discovery_rules/_form.html.erb

```diff
@@ -23,11 +24,21 @@
             <%= text_f f, :hostname,
                        :label_help => render('discovery_rules/template_inline').html_safe,
                        :label_help_options => {:title => _("Hostname for provisioned hosts"), :'data-placement' => 'bottom' } %>
-            <%= number_f f, :max_count, :label => _('Hosts Limit'), :help_inline => _('Maximum hosts provisioned with this rule (0 = unlimited)'), :min => 0, :max => (2**31 - 1) %>
-            <%= number_f f, :priority, :help_inline => _('Rule priority (lower integer means higher priority)'), :min => 0, :max => (2**31 - 1) %>
+            <%= counter_f f, :max_count, :label => _('Hosts Limit'), :help_inline => _('Maximum hosts provisioned with this rule (0 = unlimited)'), :min => 0, :max => max_count, 'data-soft-max': max_count %>
+            <%= counter_f f, :priority, :help_inline => _('Rule priority (lower integer means higher priority)'), step: 100, value: 0 %>
             <%= checkbox_f f, :enabled %>
           </div>
           <%= render 'taxonomies/loc_org_tabs', :f => f, :obj => @discovery_rule %>
         </div>
     <%= submit_or_cancel f %>
 <% end %>
+
+<%#
+  calling it by hand, because the tfm.numFields.initAll() set a minimum of 1,
```

Comment:
Please don't use comments in erb, deface gem does not work with it and fails with very cryptic messages. If the change needs comment, put it to github PR or redmine issue.

Reply:
sorry, removed them.